### DISCUSSION TITLE
Renovate self-hosted seems to be conflicting with Whitesource

### DISCUSSION THREAD

**Discussion by modean** [renovatebot/renovate#15805] [Request Help]
### How are you running Renovate?

Self-hosted

### Please select which platform you are using if self-hosting.

GitHub Enterprise Server

### If you're self-hosting Renovate, tell us what version of the platform you run.

GHE V3.1; Renovate Docker Image via docker.io/renovate/renovate:latest; Renovate Version 32.68.3 

### Was this something which used to work for you, and then stopped?

It used to work, and then stopped

### Describe the bug

We run a standalone Renovate installation (see above for details) using Jenkins to update NodeJS/NPM dependencies on a regular basis against GHE repos.

This worked quite some time even after we have been onboarded to use Whitesource. Since some times the Whitesource bot seems to be conflicting with the Renovate standalone bot cause it closes the PRs opened by the Renovate standalone bot without further notice. Thos PRs seem to be closed after about 24h.

This is  our Renovate config:

```
{
	"extends": ["local>ax/dev-central//renovate/config/ws-notebooks-ui"]
}
```

This is our Whitesource dot config file:

```
{
  "settingsInheritedFrom": "whitesource-config/whitesource-config@master",
  "scanSettings": {
    "baseBranches": [],
    "configMode": "LOCAL"
  },
  "remediateSettings": {
    "enableRenovate": false,
    "transitiveRemediation": true,
    "extends": ["local>ax/dev-central//renovate/config/ws-notebooks-ui"]
  }
}
```

This is our Whitesource local config file:

```
npm.resolveDependencies=true
npm.ignoreSourceFiles=true
npm.includeDevDependencies=false
npm.runPreStep=true
npm.ignoreNpmLsErrors=true
npm.ignoreScripts=false
npm.yarnProject=false
```

Is it a known issue that Renovate and Whitesource can't co-exist? Note that in the Whitesource config the built-in Renovate instance has been explicitly disabled. What could be the cause that a particular Whitesource plus Renovate version mix did work and eventually do to a Whitesource version update both are now conflicting with each other?

### Relevant debug logs

<details><summary>Logs</summary>

```
Copy/paste the relevant log(s) here, between the starting and ending backticks
```

</details>

### Have you created a minimal reproduction repository?

No reproduction repository

**Comment by viceice** [comment]
this is by design and nor supported. you need to use different user / branch prefix.

**Reply by rarkins** [reply]
The WhiteSource app should run with branchPrefix set to "whitesource-remediate/" by default. Did that change?

**Reply by rarkins** [reply]
And are you running them as separate apps or users, or using the same identity?

**Comment by modean** [comment]
> The WhiteSource app should run with branchPrefix set to "whitesource-remediate/" by default. Did that change?

The Renovate standalone installation uses `"branchPrefix": "renovate-"` and this did not change since we started using it. What the Whitesource app is using I can't tell cause it is a company wide installation on GHE that we only use but we don't maintain it.

> And are you running them as separate apps or users, or using the same identity?

The Renovate standalone installation uses its own user that is distinct from anything what the Whitesource app might be using.

**Reply by rarkins** [reply]
I think what is probably happening is that your `renovate.json` `branchPrefix` configuration is being used by WhiteSource so that they share the same branch naming and conflict with each other.

In general, we recommend against running WhiteSource and Renovate together the same way we'd recommend against running two Renovate bots on the same repo, although it would probably work OK if you weren't using a custom branch prefix.

Are you aware that WhiteSource runs Renovate too, and that you possibly don't need to run both?

In short, they can't co-exist (currently) if they use the same branchPrefix, which in practice means if you use a custom `branchPrefix` in your `renovate.json`, because WhiteSource will read that too. If you moved your `branchPrefix` setting into global config (e.g. `config.js` or Renovate env variables) then it should be skipped by WhiteSource and it will continue using its traditional branch prefix.

**Reply by modean** [reply]
> Are you aware that WhiteSource runs Renovate too, and that you possibly don't need to run both?

Sorry for not mentioning this in my original post. We are fully aware that Whitesource comes with a built-in Renovate instance BUT the hosted version we use does not support private NPM repos such as those hosted on Artifactory. Hence we continued using the standalone Renovate instance, that we were already using prior to the introduction of Whitesource. Per company policies we do have to use Whitesource so it's not an option to choose one or another.

We were told that there is work underway to roll out a new version of Whitesource that would support private NPM repos. I'd say that's the way to go if it really works as expected cause it allows us to get rid of the standalone Renovate instance.

**Reply by rarkins** [reply]
The hosted version of WhiteSource has supported private repos since sometime in 2021, using hostRules and encrypted values (different encryption key to the hosted Renovate app). I'm not sure if your company is on an older version, or is not aware of the new feature, or if there's an edge case causing yours not to work.. but it would be good to find out.
